wdl-analysis/tests/analysis/type-mismatch/source.diagnostics

```diff
@@ -4,69 +4,69 @@ error: type mismatch: expected type `Int`, but found type `String`
 10 │     Int a = "hello"
    │         -   ^^^^^^^ this is type `String`
    │         │    
-   │         this is type `Int`
+   │         this expects type `Int`
 
 error: type mismatch: expected type `String`, but found type `Int`
    ┌─ tests/analysis/type-mismatch/source.wdl:11:16
    │
 11 │     String b = 5
    │            -   ^ this is type `Int`
    │            │    
-   │            this is type `String`
+   │            this expects type `String`
 
 error: type mismatch: expected type `Array[String]`, but found type `Map[Int, String]`
    ┌─ tests/analysis/type-mismatch/source.wdl:12:23
    │
 12 │     Array[String] c = { 1: "one", 2: "two" }
    │                   -   ^^^^^^^^^^^^^^^^^^^^^^ this is type `Map[Int, String]`
    │                   │    
-   │                   this is type `Array[String]`
+   │                   this expects type `Array[String]`
 
 error: type mismatch: expected type `Array[Int]`, but found type `Array[String]`
    ┌─ tests/analysis/type-mismatch/source.wdl:13:20
    │
 13 │     Array[Int] d = ["a", "b", "c"]
    │                -   ^^^^^^^^^^^^^^^ this is type `Array[String]`
    │                │    
-   │                this is type `Array[Int]`
+   │                this expects type `Array[Int]`
 
 error: type mismatch: expected type `Map[Int, String]`, but found type `Map[String, Int]`
    ┌─ tests/analysis/type-mismatch/source.wdl:14:26
    │
 14 │     Map[Int, String] e = { "a": 1, "b": 2, "c": 3 }
    │                      -   ^^^^^^^^^^^^^^^^^^^^^^^^^^ this is type `Map[String, Int]`
    │                      │    
-   │                      this is type `Map[Int, String]`
+   │                      this expects type `Map[Int, String]`
 
 error: type mismatch: expected type `Int`, but found type `String`
    ┌─ tests/analysis/type-mismatch/source.wdl:15:24
    │
 15 │     Array[Int] f = [1, "2", "3"]
    │                     -  ^^^ this is type `String`
    │                     │   
-   │                     this is type `Int`
+   │                     this expects type `Int`
```

Comment:
Up until this error, switching `is` -> `expects` makes for clearer diagnostics IMO, but it gets wonky from here down. Can we treat these cases separately?

Reply:
I can further specialize this error.